parser/src/token.rs

```diff
@@ -412,6 +428,22 @@ impl<'input> Tokenizer<'input> {
                 let (end, float) = self.take_while(start, is_digit);
                 (start, end, Token::FloatLiteral(float.parse().unwrap()))
             }
+            Some((end, 'x')) => {
+                self.bump(); // Skip 'x'
+                let (end, hex) = self.take_while(end.shift('x'), is_hex);
```

Comment:
This needs a check that the next character is not an identifier otherwise `0x1q` will be tokenized as `0x1 q` . This is done later in the function for normal (decimal) integers but is actually also bugged for floats! That is, `1.0q` is accepted today. If you want it would be great if you can add the same check for floats as well  😅